include/vulkan/utility/vk_format_utils.h

```diff
@@ -0,0 +1,2268 @@
+        case VK_FORMAT_G16_B16R16_2PLANE_422_UNORM:
+        case VK_FORMAT_G16_B16_R16_3PLANE_444_UNORM:
+        case VK_FORMAT_G16_B16R16_2PLANE_444_UNORM:
+        case VK_FORMAT_R16G16_S10_5_NV:
+            return true;
+        default:
+            return false;
+    }
+}
+inline bool FormatIs32bit(VkFormat format) {
+    switch (format) {
+        case VK_FORMAT_R32_UINT:
+        case VK_FORMAT_R32_SINT:
+        case VK_FORMAT_R32_SFLOAT:
+        case VK_FORMAT_R32G32_UINT:
+        case VK_FORMAT_R32G32_SINT:
+        case VK_FORMAT_R32G32_SFLOAT:
+        case VK_FORMAT_R32G32B32_UINT:
+        case VK_FORMAT_R32G32B32_SINT:
+        case VK_FORMAT_R32G32B32_SFLOAT:
+        case VK_FORMAT_R32G32B32A32_UINT:
+        case VK_FORMAT_R32G32B32A32_SINT:
+        case VK_FORMAT_R32G32B32A32_SFLOAT:
+        case VK_FORMAT_D32_SFLOAT:
+            return true;
+        default:
+            return false;
+    }
+}
+inline bool FormatIs64bit(VkFormat format) {
+    switch (format) {
+        case VK_FORMAT_R64_UINT:
+        case VK_FORMAT_R64_SINT:
+        case VK_FORMAT_R64_SFLOAT:
+        case VK_FORMAT_R64G64_UINT:
+        case VK_FORMAT_R64G64_SINT:
+        case VK_FORMAT_R64G64_SFLOAT:
+        case VK_FORMAT_R64G64B64_UINT:
+        case VK_FORMAT_R64G64B64_SINT:
+        case VK_FORMAT_R64G64B64_SFLOAT:
+        case VK_FORMAT_R64G64B64A64_UINT:
+        case VK_FORMAT_R64G64B64A64_SINT:
+        case VK_FORMAT_R64G64B64A64_SFLOAT:
+            return true;
+        default:
+            return false;
+    }
+}
+
+inline bool FormatHasComponentSize(VkFormat format, uint32_t size) {
+    const struct FORMAT_UTILS_FORMAT_INFO format_info = GetFormatInfo(format);
+    bool equal_component_size = false;
+    for (size_t i = 0; i < FORMAT_UTILS_MAX_COMPONENTS; i++) {
+        equal_component_size |= format_info.components[i].size == size;
+    }
+    return equal_component_size;
+}
+
+inline bool FormatHasComponentType(VkFormat format, enum FORMAT_UTILS_COMPONENT_TYPE component) {
+    const struct FORMAT_UTILS_FORMAT_INFO format_info = GetFormatInfo(format);
+    bool equal_component_type = false;
+    for (size_t i = 0; i < FORMAT_UTILS_MAX_COMPONENTS; i++) {
+        equal_component_type |= format_info.components[i].type == component;
+    }
+    return equal_component_type;
+}
+
+inline bool FormatHasRed(VkFormat format) { return FormatHasComponentType(format, FORMAT_UTILS_COMPONENT_TYPE_R); }
+
+inline bool FormatHasGreen(VkFormat format) { return FormatHasComponentType(format, FORMAT_UTILS_COMPONENT_TYPE_G); }
+
+inline bool FormatHasBlue(VkFormat format) { return FormatHasComponentType(format, FORMAT_UTILS_COMPONENT_TYPE_B); }
+
+inline bool FormatHasAlpha(VkFormat format) { return FormatHasComponentType(format, FORMAT_UTILS_COMPONENT_TYPE_A); }
+
+inline uint32_t GetPlaneIndex(VkImageAspectFlagBits aspect) {
+    switch (aspect) {
+        case VK_IMAGE_ASPECT_PLANE_0_BIT:
+            return 0;
+            break;
```

Comment:
`break` after `return`?

Reply:
Originally it didn't return from inside the switch statement, but now that it does, the break is redundant. Thanks for catching it!